code/game/objects/items/weapons/material/knives.dm

```diff
@@ -25,6 +25,21 @@
 			M = user
 		return eyestab(M,user)
 
+/obj/item/weapon/material/knife/verb/extract_shrapnel(var/mob/living/carbon/human/H)
+	set name = "Extract Shrapnel"
+	set category = "Object"
+	set src in usr
+
+	for(var/obj/item/weapon/material/shard/shrapnel/S in H.contents)
+		visible_message("<span class='notice'>[usr] starts carefully digging out some of the shrapnel in [H == usr ? "themselves" : H]...</span>")
+		to_chat(H, "<font size=3><span class='danger'>It burns!</span></font>")
+		if(do_after(H, 70))
+			S.forceMove(H.loc)
+		H.apply_damage(40, HALLOSS)
+		H.apply_damage(10, BRUTE)

```

Comment:
Shrapnel is so dangerous because it's near organs, not near the surface. I would prefer a percentile chance of causing internal bleeding/organ damage, since you're literally digging around next to your muscles and organs and bones and veins with a KNIFE. That'd still make it preferable to keeping the shrapnel in if medical is unavailable, as moving around with shrapnel could cause that anyway (and chances are if you have shrapnel you've already got both), since as I said above, a chance of either recovering or a faster death is still better than a sure, slow, and immeasurably painful death.